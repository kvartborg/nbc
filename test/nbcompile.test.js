/* eslint-env jest */
import { exec } from 'child_process'

test(`Should exit if the current directory isn't a NetBeans project`, done => {
  exec('./bin/nbcompile.sh', (err, stdout) => {
    expect(err.code).toBe(1)
    expect(stdout).toMatchSnapshot()
    done()
  })
})

test(`Should compile java code and place it in the build folder`, done => {
  process.chdir('./test/project')
  exec('../../bin/nbcompile.sh', (err, stdout) => {
    expect(err).toBe(null)
    expect(stdout).toMatchSnapshot()
    done()
  })
})
