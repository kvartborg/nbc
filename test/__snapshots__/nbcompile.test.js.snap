// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should compile java code and place it in the build folder 1`] = `
"works!
"
`;

exports[`Should exit if the current directory isn't a NetBeans project 1`] = `
"Not a netbeans project
"
`;
